docker/binaries/vtctldclient/Dockerfile

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 The Vitess Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG VT_BASE_VER=latest
+ARG DEBIAN_VER=stable-slim
+
+FROM vitess/lite:${VT_BASE_VER} AS lite
+
+FROM debian:${DEBIAN_VER}
+
+RUN apt-get update && \
+   apt-get upgrade -qq && \
+   apt-get install jq curl -qq --no-install-recommends && \
+   apt-get autoremove && \
+   apt-get clean && \
+   rm -rf /var/lib/apt/lists/*
```

Comment:
I'm assuming jq and curl are in here for debugging? I know they're in the other Dockerfiles, so I don't want to gate this change on it, but should we think about moving to distroless for most of these binaries? The only ones offhand I can think of that need Linux might be vttablet since it uses installs the mysql client.

Reply:
Eventually we should think about it.

Reply:
Those are mostly to align with the existing `vtctlclient` image. I can remove the entire `RUN` layer since I don't think any of the apt-get functionality is required for this to work if that'd be preferred.

Reply:
I think what you are proposing right now is fine. IMHO moving to distroless should eventually be done to all images in a subsequent PR so we have something consistent.

Reply:
Agreed, this PR is great as is to stay consistent with vtctlclient